### PR TITLE
feat(core): organization jit sso

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

- implement provisioning by sso connectors for organization jit
- update triggers for jit

## Testing

integration tests added / updated

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [x] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/libraries/user.utils.ts
Comment (gao-sun) [inline packages/core/src/libraries/user.utils.ts:1]: this file is only a copy-paste

File: packages/core/src/libraries/user.ts [hunk 1]
```diff
@@ -338,6 +241,66 @@ export const createUserLibrary = (queries: Queries) => {
...
+  /**
+   * Provision the user with JIT organizations and roles based on the user's email domain and the
+   * enterprise SSO connector.
+   */
+  const provisionOrganizations = async ({
+    userId,
+    email,
+    ssoConnectorId,
+  }: ProvisionOrganizationsParams): Promise<readonly JitOrganization[]> => {
+    const userEmailDomain = email?.split('@')[1];
```
Comment (wangsijie) [inline packages/core/src/libraries/user.ts:272]: nit: use a util function for better debugging on error.

**Review by wangsijie — APPROVED** [review]
LGTM

File: packages/core/src/libraries/user.ts (hunk 1 above)
Comment (gao-sun) [reply packages/core/src/libraries/user.ts:272]: can you elaborate this?

Comment (gao-sun) [reply packages/core/src/libraries/user.ts:272]: discussed offline

**Pull request merged**